app/assets/javascripts/Components/Helpers/range_selector.js

```diff
@@ -54,12 +54,63 @@ function getLeafNodes(root, _nodes) {
   return _nodes;
 }
 
-export function markupTextInRange(range, colour) {
+export function pathToNode(node) {
+  if (node.id) {
+    return `//*[@id="${node.id}"]`;
+  } else if (node.parentNode) {
+    if (node.nodeType === 3) {
+      return `${pathToNode(node.parentNode)}/text()`;
+    } else {
+      const index = [...node.parentNode.childNodes]
+        .filter(n => n.tagName === node.tagName)
+        .indexOf(node);
+      return `${pathToNode(node.parentNode)}/${node.tagName}[${index + 1}]`;
+    }
+  } else {
+    return "";
+  }
+}
+
+function addMouseOverToNode(node, content) {
+  const content_span = node.ownerDocument.createElement("span");
+  console.log(document.documentElement.style.getPropertyValue("--radius"));
+  Object.assign(content_span.style, {
+    display: "none",
+    // copy of the css properties for .annotation_text_display elements
+    background: document.documentElement.style.getPropertyValue("--background_main"),
+    border: `1px solid ${document.documentElement.style.getPropertyValue("--sharp_line")}`,
+    borderRadius: document.documentElement.style.getPropertyValue("--radius"),
+    boxShadow: `4px 4px 2px ${document.documentElement.style.getPropertyValue("--primary_two")}`,
+    maxWidth: "400px",
+    padding: "0.25em 1em",
+    position: "absolute",
+    width: "auto",
+    wordWrap: "break-word",
+    zIndex: "100000",
+  });
+  content_span.innerText = content;
```

Comment:
One thing I noticed actually was that you get mathjax for free when using jupyter notebooks because jupyter itself uses mathjax to render some of the content (with a different configuration than our probably). Once we get mathjax working later on it might be a bit of a battle between our mathjax configuration and jupyter's. Just something for me/you/whoever works on this to be aware of.